pkg/actors/internal/placement/client/client.go

```diff
@@ -91,20 +100,29 @@ func New(opts Options) (*Client, error) {
 		)
 	}
 
+	var rr connector.Interface
 	if len(addresses) == 1 && strings.HasPrefix(addresses[0], "dns:///") {
 		// In Kubernetes environment, dapr-placement headless service resolves multiple IP addresses.
 		// With round robin load balancer, Dapr can find the leader automatically.
 		gopts = append(gopts, grpc.WithDefaultServiceConfig(grpcServiceConfig))
+		rr, err = roundrobin.New(roundrobin.Options{
```

Comment:
Using round robin connector to be predicated on the fact that we are in Kubernetes (I.e. a passed down bool, rather than magic string match?)